Simplified docs (#155)
LucidtechAI/cradl-docs: docs/get-started/train-high-accuracy-models-on-large-datasets.mdx

---
sidebar_position: 2
title: Train high accuracy models on large datasets
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Train high accuracy models on large datasets

> For best results we recommend training your model on as many documents as possible. The benefits of training on large 

Reviewer: ```suggestion
For best results we recommend training your model on as many documents as possible. The main benefit of training on large 
```

datasets is greater overall accuracy. Cradl is built to scale with large datesets, and it is not uncommon to train on 
millions of documents. Our unique training algorithm makes training on large datasets neither time-consuming nor 

> complicated, and will not have any impact on the processing speed of your model. If you are already processing 

Reviewer: ```suggestion
complicated, and will not have any impact on the processing speed of your final model. If you are already processing 
```

documents today, chances are that you have the data needed to train a high accuracy model in Cradl.

## Step 1: Download documents with metadata from your database

*Goal: Download documents with metadata from your database and transform them to a Cradl compatible format*

<br/>

> In order to train a model in Cradl we need documents and annotations. For each document we need its corresponding 

Reviewer: ```suggestion
To train a model in Cradl we need documents and annotations. Each document must have a corresponding 
```

> annotations. These are key-value pairs containing the pieces of information which we are teaching the model to extract. 

Reviewer: ```suggestion
annotation. An annotation is a collection of key-value pairs containing the pieces of information we are teaching the model to extract. 
```

> Notice that **Cradl does not require you to provide any positional metadata such as bounding box coordinates**, we are 

Reviewer: ```suggestion
Notice that **Cradl does not require you to provide any positional metadata such as bounding box coordinates**. We are 
```

only interested in key-value pairs such as `total_amount=100.00` or `invoice_date=2021-10-30`. Annotations are 

> formatted in JSON. Here is an example of a document and its corresponding annotations.

Reviewer: ```suggestion
formatted in JSON. Below is an example of a document and its corresponding annotations.
```

<br/>

![Document and json ground truth example](/img/get-started/document-and-json-ground-truth-example.png)

To make the documents and annotations ready for upload in Cradl we need to have all of them in a directory with the 

> following constraint: for each document and its annotation file they must share the same filename prefix. See example 

Reviewer: ```suggestion
following constraint: a document and its annotation file must have the same filename prefix. See the example 
```

below.

![Documents and ground truth folder example](/img/get-started/documents-and-ground-truth-folder.svg)

:::tip
Having good training data is important in order to get high accuracy. Read more about training data 
[here](../concepts/training-data#data-quality).
:::

## Step 2: Upload data to Cradl

*Goal: Upload your documents and annotations to Cradl for training*

<br/>

> Before we can begin uploading documents and annotations you need to have initiated a training job wizard on your model 

Reviewer: ```suggestion
Before uploading documents and annotations you need to initiate a training job wizard for your model 
```

<TabItem value="WebApp" label="Using the WebApp" default>

Drag all the files into the drop-zone as shown in below illustration.

![Dropzone example](/img/get-started/dropzone-example.png)

</TabItem>

<TabItem value="CLI" label="Using the CLI" default>

:::info
You need to have the CLI installed. Follow the instructions [here](../reference/cli) to install the CLI.
:::

Copy the `datasetId` from the WebApp as shown in the below illustration.

![Copy datasetId](/img/get-started/copy-dataset-id.png)

Then use the CLI to upload documents and annotations from your folder to this dataset. 

```bash
$ las datasets create-documents <your datasetId> <your folder with documents and annotations>
```

</TabItem>

</Tabs>

## Step 3: Review your data and start training

*Goal: Make sure the data looks alright and start training the model*

<br/>

You should now see the documents you uploaded in the previous step appearing in the WebApp. To proceed, follow the 
instructions in the WebApp and you will be taken to a view where you are able to edit or create new annotations for 
your documents. If you have already successfully uploaded annotations in the previous step and don't need to add any 

> new ones, then you probably don't need to do anything here. Proceed to the last step where you will get a statistical 

Reviewer: ```suggestion
more, then you probably don't need to do anything here. Proceed to the last step where you will get a statistical 
```
